src/Microsoft.OpenApi/Services/OpenApiUrlSpaceNode.cs

```diff
@@ -0,0 +1,233 @@
+        /// Path Item object that describes the operations available on a node.
+        /// </summary>
+        public OpenApiPathItem PathItem { get; private set; }
+
+        /// <summary>
+        /// The relative directory path of the current node from the root node.
+        /// </summary>
+        public string Path { get; private set; } = "";
+
+        /// <summary>
+        /// Flag indicating whether a node segment is a path parameter.
+        /// </summary>
+        public bool IsParameter => Segment.StartsWith("{");
+
+        /// <summary>
+        /// Flag indicating whether a node segment is a path function.
+        /// </summary>
+        public bool IsFunction => Segment.Contains("(");
+
+        /// <summary>
+        /// The subdirectory of a relative path.
+        /// </summary>
+        public string Segment { get; private set; }
+
+        /// <summary>
+        /// The Pascal-cased alphabet name of a segment.
+        /// </summary>
+        public string Identifier
+        {
+            get
+            {
+                string identifier;
+                if (IsParameter)
+                {
+                    identifier = Segment.Substring(1, Segment.Length - 2).ToPascalCase();
+                }
+                else
+                {
+                    identifier = Segment.ToPascalCase().Replace("()", "");
+                    var openParen = identifier.IndexOf("(");
+                    if (openParen >= 0)
+                    {
+                        identifier = identifier.Substring(0, openParen);
+                    }
+                }
+                return identifier;
+            }
+        }
+
+        /// <summary>
+        /// Flag indicating whether a PathItem has operations.
+        /// </summary>
+        /// <returns>true or false.</returns>
+        public bool HasOperations()
+        {
+            return PathItem?.Operations != null && PathItem.Operations.Count > 0;
+        }
+
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="segment">The subdirectory of a relative path.</param>
+        public OpenApiUrlSpaceNode(string segment)
+        {
+            Segment = segment;
+        }
+
+        /// <summary>
+        /// Uses SHA256 hash algorithm to hash the Path value.
+        /// </summary>
+        /// <returns>The hashed value.</returns>
+        public string Hash()
+        {
+            using (SHA256 sha256Hash = SHA256.Create())
+            {
+                return GetHash(sha256Hash, Path);
+            }
+        }
+
+        /// <summary>
+        /// Hashes a string value using a specified hash algorithm.
+        /// </summary>
+        /// <param name="hashAlgorithm">The hash algorithm to use for hashing.</param>
+        /// <param name="input">The string to hash.</param>
+        /// <returns>The hashed value.</returns>
+        private static string GetHash(HashAlgorithm hashAlgorithm, string input)
```

Comment:
This function is related to us generating unique identifiers.  I don't believe this should be in this library.  This is Kiota related.